packages/dx-react-chart/src/templates/axis/root.jsx

```diff
@@ -26,18 +25,15 @@ export class Root extends React.PureComponent {
     this.setState(this.adjust); // eslint-disable-line react/no-did-update-set-state
   }
 
-  adjust({ width: prevWidth, height: prevHeight }, { dx, dy, onSizeChange }) {
+  adjust(_, { dx, dy, onSizeChange }) {
     const bbox = this.node.getBBox();
     const width = dx ? bbox.width : getSize(bbox.x, bbox.width);
     const height = dy ? bbox.height : getSize(bbox.y, bbox.height);
     const x = dx ? 0 : getOffset(bbox.x);
     const y = dy ? 0 : getOffset(bbox.y);
-    if (width === prevWidth && height === prevHeight) {
-      return null;
-    }
     onSizeChange({ width, height });
     return {
```

Comment:
Since now new state does not depend on current state, callback version of `setState` is not required.
This line can then be replaced with `this.setState({ x, y })` and in `componentDidMount` and `componentDidUpdate` those single lines can be resplaced with `this.adjust()`;
There is comment over `componentDidUpdate` - it should be update. Now unconditional `setState` relies on the "pureness" of the component.